mixer/adapter/redisquota/redisquota.go

```diff
@@ -0,0 +1,400 @@
+				continue
+			}
+
+			if _, err := getDimensionHash(b.adapterConfig.Quotas[idx].Overrides[index].Dimensions); err != nil {
+				ce = ce.Appendf(info.Name, "unable to initialize quota overrides dimensions",
+					b.adapterConfig.Quotas[idx].Overrides[index].Dimensions)
+			}
+		}
+	}
+
+	for k := range b.quotaTypes {
+		if _, ok := limits[k]; !ok {
+			ce = ce.Appendf(info.Name, "did not find limit defined for quota %v", k)
+		}
+	}
+
+	// check redis related configuration
+	if b.adapterConfig.ConnectionPoolSize < 0 {
+		ce = ce.Appendf(info.Name, "connection_pool_size of %v is invalid, must be > 0",
+			b.adapterConfig.ConnectionPoolSize)
+	}
+
+	if len(b.adapterConfig.RedisServerUrl) == 0 {
+		ce = ce.Appendf(info.Name, "redis_server_url should not be empty")
+	}
+
+	// test redis connection
+	option := redis.Options{
+		Addr: b.adapterConfig.RedisServerUrl,
+	}
+
+	if b.adapterConfig.ConnectionPoolSize > 0 {
+		option.PoolSize = int(b.adapterConfig.ConnectionPoolSize)
+	}
+
+	client := redis.NewClient(&option)
+	if _, err := client.Ping().Result(); err != nil {
+		ce = ce.Appendf(info.Name, "could not create a connection to redis server: %v", err)
+		return
+	}
+
+	// check scripts loading to redis
+	scripts := make(map[config.Params_QuotaAlgorithm]*redis.Script, 2)
+	for algorithm, script := range rateLimitingLUAScripts {
+		scripts[algorithm] = redis.NewScript(script)
+		if _, err := scripts[algorithm].Load(client).Result(); err != nil {
+			ce = ce.Appendf(info.Name, "unable to initialized redis service: %v", err)
+			return
+		}
+	}
+
+	_ = client.Close()
+	return
+}
+
+// getOverrideHash returns hash key of the given dimension in sorted by key
+func getDimensionHash(dimensions map[string]string) (string, error) {
+	var keys []string
+	for k := range dimensions {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	h := fnv.New32a()
+	for _, key := range keys {
+		if _, err := io.WriteString(h, key+"\t"+dimensions[key]+"\n"); err != nil {
+			return "", err
+		}
+	}
+	return strconv.Itoa(int(h.Sum32())), nil
+}
+
+func (b *builder) Build(context context.Context, env adapter.Env) (adapter.Handler, error) {
+	limits := make(map[string]*config.Params_Quota, len(b.adapterConfig.Quotas))
+	for idx := range b.adapterConfig.Quotas {
+		limits[b.adapterConfig.Quotas[idx].Name] = &b.adapterConfig.Quotas[idx]
+	}
+
+	// Build memory address of dimensions to hash map
+	dimensionHash := make(map[*map[string]string]string)
+	for key := range limits {
+		for index := range limits[key].Overrides {
+			if hash, err := getDimensionHash(limits[key].Overrides[index].Dimensions); err == nil {
+				dimensionHash[&(limits[key].Overrides[index].Dimensions)] = hash
+			}
+		}
+	}
+
+	// initialize redis client
+	option := redis.Options{
+		Addr: b.adapterConfig.RedisServerUrl,
+	}
+
+	if b.adapterConfig.ConnectionPoolSize > 0 {
+		option.PoolSize = int(b.adapterConfig.ConnectionPoolSize)
+	}
+
+	client := redis.NewClient(&option)
+
```

Comment:
should we ping the client and check again (though we have already check on validate, since validate and build are called independently) before returning from this method.

Reply:
Added additional Ping test